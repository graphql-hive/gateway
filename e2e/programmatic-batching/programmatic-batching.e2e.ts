import { createTenv } from '@internal/e2e';
import { expect, it } from 'vitest';

const { service, gateway } = createTenv(__dirname);

it.concurrent.each([
  {
    name: 'User',
    query: /* GraphQL */ `
      fragment UserF on User {
        id
        name
      }
      query User {
        john: user(id: 1) {
          ...UserF
        }
        jane: user(id: 2) {
          ...UserF
        }
      }
    `,
  },
])('should execute $name', async ({ query }) => {
  const { execute } = await gateway({
    supergraph: {
      with: 'mesh',
      services: [await service('api')],
    },
  });
  await expect(execute({ query })).resolves.toMatchSnapshot();
});
