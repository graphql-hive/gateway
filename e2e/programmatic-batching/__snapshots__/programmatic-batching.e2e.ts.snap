// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should execute 'User' 1`] = `
{
  "data": {
    "jane": {
      "id": 2,
      "name": "Jane Doe",
    },
    "john": {
      "id": 1,
      "name": "John Doe",
    },
  },
}
`;
